### PR TITLE
Update to v0.22.0

### LINKED ISSUE #155: regro-cf-autotick-bot erroring with `v0.22` update
### Comment:

I know you've all been working hard on release automation 🤩, so I just wanted to make you aware that the PyPI-to-conda automation isn't currently working.

Cartopy appears in the [conda-forge version updates list](https://conda-forge.org/status/#version_updates) under the `Errored` heading, with this text:

```
2.00 attempts - The recipe did not change in the version migration, a URL did not hash, or there is jinja2 syntax the bot cannot handle!

Please check the URLs in your recipe with version '0.22.0' to make sure they exist!

We also found the following errors:

 - could not hash URL template 'https://pypi.io/packages/source/C/Cartopy/Cartopy-{{ version }}.tar.gz'
```

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes #155

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
I would appreciate at least one more set of eyes on:
https://github.com/SciTools/cartopy/blob/v0.22.0/pyproject.toml
to make sure I didn't mess anything up.

It would also be worth discussing if we want to make `cartopy-with-*` packages with any of the optional dependencies:
https://github.com/SciTools/cartopy/blob/v0.22.0/pyproject.toml#L52-L56
My current inclination is not to do that unless there is demand but it wouldn't be hard.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,44 +1,41 @@
...
 
 package:
   name: cartopy
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/C/Cartopy/Cartopy-{{ version }}.tar.gz
-  sha256: 89d5649712c8582231c6e11825a04c85f6f0cee94dbb89e4db23eabca1cc250a
+  url: https://github.com/SciTools/cartopy/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 7adf6b92f4e16c46fb1ce58840275af7a6dc1f21d5243d4b665ecef72961f9f4
```
Comment (xylar) [inline recipe/meta.yaml:9]: Source is no longer being included on PyPI as discussed in #155

**Comment by xylar** [comment]
```
      LookupError: setuptools-scm was unable to detect version for /home/conda/feedstock_root/build_artifacts/cartopy_1691225778495/work.
      
      Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.
      
      For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
      [end of output]
```
I will try to remind myself what the workaround for this is but in the meantime I've submitted an issue upstream:
https://github.com/SciTools/cartopy/issues/2228

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,44 +1,41 @@
...
-    - shapely >=1.6.4
+    - matplotlib-base >=3.4
+    - shapely >=1.7
+    - packaging >=20
     - pyshp >=2.1
-    - pyproj >=3.0.0
-    - geos
-    - matplotlib-base >=3.1
-    - scipy >=0.10
-    - ucrt  # [win]
```
Comment (xylar) [inline recipe/meta.yaml:41]: Scipy is only included as an extra in this version.  I'm not sure what `ucrt` was needed for but I didn't see any mention of it in this release.

**Comment by xylar** [comment]
@conda-forge/cartopy, this is ready for review.

**Comment by xylar** [comment]
A question is whether we're okay with switching to GitHub for the source as done here or if we want to wait for source to be uploaded to PyPI, see https://github.com/SciTools/cartopy/issues/2228#issuecomment-1666504204

**Comment by ocefpaf** [comment]
> A question is whether we're okay with switching to GitHub for the source as done here or if we want to wait for source to be uploaded to PyPI, see [SciTools/cartopy#2228 (comment)](https://github.com/SciTools/cartopy/issues/2228#issuecomment-1666504204)

I'm OK moving to GH and even moving back to PyPI after this release. Whatever you feel is easier. The only advantage to PyPI releases is that, some times, they are slimmer than the GH counterparts b/c authors remove some of the "fat."

**Comment by xylar** [comment]
@ocefpaf, I also feel like bypassing `setuptools-scm` isn't very clean so I would prefer to see if the source gets pushed in the next day or 2 as it seems it might.

**Comment by dopplershift** [comment]
I just uploaded the sdist to PyPI.

**Comment by xylar** [comment]
Thanks @dopplershift!

**Comment by xylar** [comment]
@conda-forge/cartopy, I have moved the source back to pypi and this is again ready for review (assuming tests pass).

File: recipe/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipe/meta.yaml:41]: It is a universal C runtime and it was added in https://github.com/conda-forge/cartopy-feedstock/pull/116#issuecomment-912565452

The CIs are passing so I'm OK leaving it out. Hopefully we won't break cartopy for Windows users :-)

**Review by ocefpaf — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (xylar) [reply recipe/meta.yaml:41]: @zklaus, you were the one to recommend adding this library.  It seemed to make CI pass at the time but doesn't seem to be needed now.  I didn't find any mention of it (outside of people listing their environments) upstream: https://github.com/search?q=repo%3ASciTools%2Fcartopy+ucrt&type=issues

Do you think it's okay to remove it at this point?

Comment (akrherz) [reply recipe/meta.yaml:41]: IIRC, @zklaus is on holiday this month

Comment (xylar) [reply recipe/meta.yaml:41]: Ah, yes.  In that case maybe we assume the passing tests are a sign that this is no longer needed?  It seemed like the PR where this was added indicated that the Win tests only passed after it was added.

Comment (dopplershift) [reply recipe/meta.yaml:41]: I say merge and we can always add it back if someone reports it broken. Better to get this out there.

**Review by dopplershift — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (dopplershift) [reply recipe/meta.yaml:41]: I'll add that cartopy isn't doing anything special here on building (just a bit of Cython at this point), so if it is somehow needed, it really should be something else exporting the requirement.

**Pull request merged**